pkg/healthcheck/healthcheck.go

```diff
@@ -84,7 +85,7 @@ func waitForConsole(ctx context.Context, cs *acsapi.ContainerService, c *config.
 		Timeout: 10 * time.Second,
 	}
 
-	req, err := http.NewRequest("HEAD", "https://"+cs.Properties.OrchestratorProfile.OpenShiftConfig.PublicHostname+"/console/", nil)
```

Comment:
This is not quite right. But if PublicHostname is provided by user and not set by RP this test fails.

Reply:
I think you can do this by using a custom HTTP Transport (see https://golang.org/pkg/net/http/#RoundTripper) and overriding Dial to dial the PublicHostname